plugins/ImageGraph/API.php

```
@@ -129,404 +130,453 @@ public function get(
-        $languageLoaded = Translate::getLanguageLoaded();
-        $font = self::getFontPath(self::DEFAULT_FONT);
-        if (in_array($languageLoaded, $useUnicodeFont)) {
-            $unicodeFontPath = self::getFontPath(self::UNICODE_FONT);
-            $font = file_exists($unicodeFontPath) ? $unicodeFontPath : $font;
-        }
-
-        // save original GET to reset after processing. Important for API-in-API-call
-        $savedGET = $_GET;
-
-        try {
-            $apiParameters = array();
-            if (!empty($idGoal)) {
-                $apiParameters = array('idGoal' => $idGoal);
+        $idSiteArr = array();
+        if (strpos($idSite, ',') !== false) {
+            $idSiteList = explode(',', $idSite);
+            foreach ($idSiteList as $key) {
+                $idSiteArr[] = $key;
             }
-            // Fetch the metadata for given api-action
-            $parameters = array(
-                'idSite' => $idSite,
-                'apiModule' => $apiModule,
-                'apiAction' => $apiAction,
-                'apiParameters' => $apiParameters,
-                'language' => $languageLoaded,
-                'period' => $period,
-                'date' => $date,
-                'hideMetricsDoc' => false,
-                'showSubtableReports' => true
-            );
-
-            $metadata = Request::processRequest('API.getMetadata', $parameters);
-            if (!$metadata) {
-                throw new Exception('Invalid API Module and/or API Action');
+        } else if (!empty($idSite) && ($idSite != 'all')) {
+            $idSiteArr[] = $idSite;
+        } else if (!empty($idSite) && ($idSite == 'all')) {
+            $viewableIdSites = APISitesManager::getInstance()->getSitesIdWithAtLeastViewAccess();
+            foreach ($viewableIdSites as $site) {
+                $idSiteArr[] = $site;
             }
+        } else {
+            // empty or invalid idSite has been already handled somewhere
+            exit;
```

We usually don't to `exit` in our code. Can we instead throw an exception or return?